aws/service_v2.go

```diff
@@ -1250,3 +1325,65 @@ func (j *ExponentialJitterBackoff) BackoffDelay(attempt int, err error) (time.Du
 
 	return retryTime, nil
 }
+
+// SupportedRegionsForClient list outs the valid regions for a service based on service id
+func GetSupportedRegionsForClient(ctx context.Context, d *plugin.QueryData, serviceId string) ([]string, error) {
+	var partitionName string
+	var partition endpoints.Partition
+
+	// If valid regions list is already avialable in cache, return it
+	cacheKey := fmt.Sprintf("supported-regions-%s", serviceId)
+	if cachedData, ok := d.ConnectionManager.Cache.Get(cacheKey); ok {
+		return cachedData.([]string), nil
+	}
+
+	// Get the partition of the AWS account plugin is connected to
+	if cachedData, ok := d.ConnectionManager.Cache.Get("getAccountPartition"); ok {
+		partitionName = cachedData.(string)
+	}
+
+	// If partition name is not available,
+	// try to fetch partiton from APIs
+	if partitionName == "" {
+		// Using WithCache to avoid making repeated calls to fetch partition in parallel
+		getCachedAccountPartition := plugin.HydrateFunc(getAccountPartition).WithCache()
+		// Ignoring the errors as we don't want plugin to fail if unable to determine the partitions
+		// Instead below switch statment will set AWS commercial as the default partition
+		partitionData, _ := getCachedAccountPartition(ctx, d, nil)
+		partitionName, _ = partitionData.(string)
+	}
+
+	// Get AWS partition based on the partition name
+	// If the partition name was obtained from APIs set AWS commercial as the default partition
```

Comment:
Do we only set AWS commercial as the default if it was obtained from APIs? It seems like we're always using that as the default here